server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/reactivestreams/SimpleSubscriber.scala

```diff
@@ -5,15 +5,16 @@ import io.netty.handler.codec.http.HttpContent
 import org.reactivestreams.{Publisher, Subscription}
 import sttp.capabilities.StreamMaxLengthExceededException
 
-import java.util.concurrent.{ConcurrentLinkedQueue, LinkedBlockingQueue}
+import java.util.concurrent.LinkedBlockingQueue
+import scala.collection.mutable
 import scala.concurrent.{Future, Promise}
 
 private[netty] class SimpleSubscriber(contentLength: Option[Int]) extends PromisingSubscriber[Array[Byte], HttpContent] {
   private var subscription: Subscription = _
   private val resultPromise = Promise[Array[Byte]]()
   private var totalLength = 0
   private val resultBlockingQueue = new LinkedBlockingQueue[Either[Throwable, Array[Byte]]]()
-  private val buffers = new ConcurrentLinkedQueue[ByteBuf]()
+  private val buffers = new mutable.ListBuffer[ByteBuf]()
```

Comment:
follow-up question (sorry ;) ) - onNext/onComplete/onError are guaranteed to be called from one thread, but is it going to be the same thread? that is, does `buffers` need to be `volatile`?

Reply:
We don't have such a guarantee, and maybe that was why I had a `ConcurrentLinkedQueue` for byte arrays in the previous implementation, but I forgot :) This means I either fall back to it, or use a `volatile ListBuffer`.
I guess `var totalLength` is also unsafe and should be converted to an `AtomicInteger`, right?

Reply:
Update: You can't have a `volatile val`, so I fell back to `ConcurrentLinkedQueue`. For `totalLength` I chose `@volatile` instead of `AtomicInteger`, because we are using this variable sequentially, our only scenario is increasing it in `onNext` and reading in `onComplete`.

Reply:
but we can have a volatile var with an immutable list - always less synchronisations

Reply:
Ok, I replaced the `ConcurrentLinkedQueue` with a volatile `var ListBuffer`. In theory, `ListBuffer` should have slightly cheaper append time than a `Vector`, and we don't need Vector's fast random access, which takes additional cost of maintaining more complex underlying structure.
I don't see any improvement of throughput, but there's a slight improvement in latency.

Reply:
I'm not sure this properly protects the value - `ListBuffer` is mutable, and the `volatile` only ensures there's a memory barrier before reading the `buffers` reference (not references inside the `ListBuffer`). But maybe since we have a memory barrier, everything will be synchronized correctly ... as you can't really access the inner references before first reading the `buffers` reference (which creates the barrier).

Anyway, I though about a simpler design, using immutable data structures, where you don't have to think that much ;-) But maybe this one works as well :)